plugins/bil/bil_main.ml

```diff
@@ -34,44 +33,46 @@ let eliminate_dead_code bil =
       | st -> loop (st :: acc) bil in
   loop [] bil
 
-let apply bil =
-  List.fold ~init:bil ~f:(fun bil f -> f bil)
-
-let norml level =
-  if level = 0 then ident
-  else
-    Stmt.normalize ~keep_ites:false ~normalize_exp:false
+let () =
+  Bil.register_pass "normalization" @@
+  Stmt.normalize ~keep_ites:false ~normalize_exp:false
 
-let simpl level =
-  if level = 0 then ident
-  else
-    Bil.fixpoint
-      (fun bil -> Bil.fold_consts bil |> Bil_const_propagation.run |> eliminate_dead_code)
+let () =
+  Bil.register_pass "optimization" @@
+  Bil.fixpoint
+    (fun bil -> Bil.fold_consts bil |> Bil_const_propagation.run |> eliminate_dead_code)
 
-let run norml_level simpl_level =
-  provide_bil_transformation "internal" @@
-  fun bil -> apply bil [
-    norml norml_level;
-    simpl simpl_level;
-  ]
+let run passes =
+  let passes = ["normalization"; "optimization"] @ passes in
```

Comment:
nay, you shall keep the  normalization and optimization parameters. The selected set of passes should be 
```ocaml
norm n @ optim m @ passes
```
where `norm n` is the list of normalizations for level `n`, `optim m` is the list of optimizations on the optimization level `m`, and `passes` are the passes that are selected by a user.

This interface allows disabling both optimization and normalization, as well as an option to define arbitrary pipeline, e.g., `--bil-normalization=0 --bil-optimization=0 --bil-passes=my-normalization,my-optimization`